'''
=================================================
Graded Challenge 3

Nama  : ROSALIA

Batch : REMOTE 053

This program is created to conduct web scraping of house price data in Bekasi.
The data is taken from Rumah123, followed by data preparation and statistical analysis.
The results are used to support analysis and decision making when purchasing a house.

=================================================

'''

# A. Web Scraping

Lakukan pengambilan data dan filter untuk beli rumah di daerah "Bekasi". Anda bisa langsung akses link ini:

https://www.rumah123.com/jual/bekasi/rumah/.

In [2]:
from selenium import webdriver

In [4]:
driver = webdriver.Chrome()
driver.get("https://www.rumah123.com/jual/bekasi/rumah/")

In [29]:
import time
from bs4 import BeautifulSoup
from selenium import webdriver


driver = webdriver.Chrome()

url = "https://www.rumah123.com/jual/bekasi/rumah/?page=1"
driver.get(url)

html = driver.page_source
soup = BeautifulSoup(html, "html.parser")

print(soup.prettify()[:700])

<html data-release="be7151d298101f8a6c477f900a910ec2f726c79a" lang="id-ID">
 <head>
  <meta charset="utf-8"/>
  <meta content="width=device-width, initial-scale=1, minimum-scale=1, maximum-scale=5, user-scalable=yes" name="viewport"/>
  <link as="image" href="https://www.rumah123.com/asset-core/images/lang-switcher-mobile.gif" rel="preload"/>
  <link as="image" fetchpriority="high" imagesizes="(max-width: 480px) 100vw, (max-width: 768px) 100vw, (max-width: 1024px) 50vw, 600px" imagesrcset="
                    https://picture.rumah123.com/r123-images/720x420-crop/customer/1971002/6df4897f6cc02ea734d2de4880bde833.jpg?w=200 200w,
                    https://picture.rumah123.com/r123-images/720


## Ambillah data mengenai :

###  
   - `Lokasi`, 
   - `Luas Tanah`, 
   - `Luas Bangunan`, 
   - `Carport`,
   - `Jumlah Kamar Tidur`, 
   - `Jumlah Kamar Mandi`, dan 
   - `Harga Rumah`.

Ambil informasi rumah minimal 10 halaman (perhatikan format URL yang digunakan dan lakukanlah eksplorasi terlebih dahulu halaman web-nya).

In [50]:
# Storing Data

lokasi_list = []
luas_tanah_list = []
luas_bangunan_list = []
carport_list = []
kamar_tidur_list = []
kamar_mandi_list = []
harga_list = []

total_pages = 10

link_url = "https://www.rumah123.com/jual/bekasi/rumah/?page="

In [51]:
# Looping : min 10 pages & 100 data
 
for page in range(1, total_pages + 1):

    print("Scraping page:", page)

    url = link_url + str(page)
    driver.get(url)
    time.sleep(5)

    html = driver.page_source
    soup = BeautifulSoup(html, "html.parser")

    house_list = soup.find_all("div", {"data-name": "ldp-listing-content"})

    print("Total houses found:", len(house_list))

Scraping page: 1
Total houses found: 20
Scraping page: 2
Total houses found: 20
Scraping page: 3
Total houses found: 20
Scraping page: 4
Total houses found: 20
Scraping page: 5
Total houses found: 20
Scraping page: 6
Total houses found: 20
Scraping page: 7
Total houses found: 20
Scraping page: 8
Total houses found: 20
Scraping page: 9
Total houses found: 20
Scraping page: 10
Total houses found: 20


### Insights : Above loop (1st looping) is needed to test the container and it has successfully been done. 

In [52]:
for page in range(1, total_pages + 1):

    print("Scraping page:", page)

    page_url = link_url + str(page)
    driver.get(page_url)
    time.sleep(5)

    html = driver.page_source
    soup = BeautifulSoup(html, "html.parser")

    house_list = soup.find_all("div", {"data-name": "ldp-listing-content"})

    for house in house_list:

        # =========================
        # LOKASI
        # =========================
        lokasi_element = house.find("p")
        if lokasi_element:
            lokasi_list.append(lokasi_element.text.strip())
        else:
            lokasi_list.append(None)

        # =========================
        # DETAIL INFO (KT, KM, Carport, LT, LB)
        # =========================
        info_box = house.find("div", class_="bg-lightBlue")

        kamar_tidur = None
        kamar_mandi = None
        carport = None
        luas_tanah = None
        luas_bangunan = None

        if info_box:

            span_items = info_box.find_all("span")

            for item in span_items:
                text = item.text.strip()

                # Bedroom
                if "+" in text and kamar_tidur is None:
                    kamar_tidur = text

                # Bathroom
                elif "+" in text and kamar_tidur is not None:
                    kamar_mandi = text

                # Carport (only number)
                elif text.isdigit():
                    carport = text

                # Luas Tanah & Luas Bangunan (both contain m²)
                elif "m²" in text:
                    if luas_tanah is None:
                        luas_tanah = text.replace("LT:", "").strip()
                    else:
                        luas_bangunan = text.replace("LB:", "").strip()

        kamar_tidur_list.append(kamar_tidur)
        kamar_mandi_list.append(kamar_mandi)
        carport_list.append(carport)
        luas_tanah_list.append(luas_tanah)
        luas_bangunan_list.append(luas_bangunan)

        # =========================
        # HARGA
        # =========================
        harga_element = house.find("span", {"data-testid": "ldp-text-price"})
        if harga_element:
            harga_list.append(harga_element.text.strip())
        else:
            harga_list.append(None)

print("Scraping DONE")

Scraping page: 1
Scraping page: 2
Scraping page: 3
Scraping page: 4
Scraping page: 5
Scraping page: 6
Scraping page: 7
Scraping page: 8
Scraping page: 9
Scraping page: 10
Scraping DONE


### Insights : Real Scraping (Second looping) here is to execute the real scrapping and it has been successfully done. 

In [53]:
print(len(harga_list))
print(len(lokasi_list))
print(len(luas_tanah_list))

200
200
200


### All scraping has been successful 
- 10 pages scraped
- 20 listings per page
- 200 total data
- All required columns collected

### Simpan hasil scraping ini ke Pandas DataFrame dan lakukanlah pengolahan terhadapnya sesuai dengan instruksi selanjutnya dibawah ini. Anda dibebaskan memberikan nama kolom yang memudahkan Anda.

## B. Data Preparation
Lakukan eksplorasi data sederhana dibawah ini :

## 1b.  Tampilkan beberapa baris data dan tuliskan insight-nya.

In [54]:
import pandas as pd

data = pd.DataFrame({
    "Lokasi": lokasi_list,
    "Luas Tanah": luas_tanah_list,
    "Luas Bangunan": luas_bangunan_list,
    "Carport": carport_list,
    "Jumlah Kamar Tidur": kamar_tidur_list,
    "Jumlah Kamar Mandi": kamar_mandi_list,
    "Harga Rumah": harga_list
})

data.head()

,Lokasi,Luas Tanah,Luas Bangunan,Carport,Jumlah Kamar Tidur,Jumlah Kamar Mandi,Harga Rumah
0,Premier,144 m²,200 m²,1,5 + 1,3 + 1,"Rp 2,2 Miliar"
1,Premier,105 m²,120 m²,1,None,None,"Rp 1,2 Miliar"
2,Premier,160 m²,215 m²,6,2 + 2,None,"Rp 6,8 Miliar"
3,Premier,112 m²,156 m²,4,2 + 2,None,"Rp 4,6 Miliar"
4,Premier,84 m²,87 m²,3,1 + 1,None,"Rp 2,3 Miliar"


# Insights : 
Berdasarkan lima baris pertama, dataset sudah memuat informasi utama properti seperti lokasi, luas, jumlah kamar, dan harga. 
- Namun masih terdapat nilai None pada beberapa kolom, yang menandakan adanya data yang belum lengkap.

1b. Tampilkan informasi rangkuman data dan tuliskan insight-nya. Berikan juga penjelasan mengenai langkah apa yang akan Anda lakukan.

In [56]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 7 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   Lokasi              200 non-null    object
 1   Luas Tanah          200 non-null    object
 2   Luas Bangunan       200 non-null    object
 3   Carport             191 non-null    object
 4   Jumlah Kamar Tidur  60 non-null     object
 5   Jumlah Kamar Mandi  16 non-null     object
 6   Harga Rumah         200 non-null    object
dtypes: object(7)
memory usage: 11.1+ KB


### Insight : 
- Dataset ini punya 200 baris dan 7 kolom.
- Semua kolom masih bertipe object karena datanya masih berupa teks (masih ada tulisan seperti “m²” dan “Rp”).
- Beberapa kolom seperti Lokasi, Luas Tanah, Luas Bangunan, dan Harga Rumah sudah lengkap (200 non-null)
- Tapi kolom Jumlah Kamar Tidur dan terutama Jumlah Kamar Mandi paling banyak yang kosong (hanya 60 & 12 terisi)

In [57]:
data.describe()

,Lokasi,Luas Tanah,Luas Bangunan,Carport,Jumlah Kamar Tidur,Jumlah Kamar Mandi,Harga Rumah
count,200,200,200,191,60,16,200
unique,1,69,67,7,11,6,89
top,Premier,60 m²,70 m²,2,1 + 1,3 + 1,Rp 610 Juta
freq,200,25,13,88,20,6,10


### Insight :
- Karena semua kolom masih berupa teks, maka yang muncul hanya ringkasan seperti jumlah data unik dan nilai yang paling sering muncul.
- Terlihat bahwa variasi luas tanah dan luas bangunan cukup beragam.
- Harga rumah yang paling sering muncul adalah Rp 610 Juta.

## 1b. Cek Missing Value dan tuliskan insight dari temuan yang Anda dapatkan.

In [58]:
data.isnull().sum()

Lokasi                  0
Luas Tanah              0
Luas Bangunan           0
Carport                 9
Jumlah Kamar Tidur    140
Jumlah Kamar Mandi    184
Harga Rumah             0
dtype: int64

## Insight : 
- Dari hasil pengecekan missing value, terlihat kolom Carport memiliki 9 data kosong.
- Kolom Jumlah Kamar Tidur memiliki 140 data kosong dan Jumlah Kamar Mandi memiliki 184 data kosong
- Hal ini menunjukkan bahwa tidak semua listing mencantumkan informasi jumlah kamar secara lengkap.

## DATA CLEANING 

- Handling Missing Value bila ada.
- Mengubah bentuk data agar menjadi lebih konsisten.
- Menyesuaikan tipe data.
- Perlu diingat bahwa data ini akan digunakan untuk perhitungan statisik dan membutuhkan kolom dengan tipe data angka.

In [ ]:
# CLEAN : HARGA RUMAH

def clean_harga(price):
    if pd.isna(price):
        return None
    
    # Ambil hanya harga pertama (sebelum Rp kedua / promo)
    first_price = price.split("Rp")[1]
    first_price = first_price.split("Rp")[0].strip()
    
    # Hilangkan kata promo jika ada
    first_price = first_price.split("HEMAT")[0].strip()
    
    if "Miliar" in first_price:
        value = float(first_price.replace("Miliar", "").replace(",", ".").strip())
        return value * 1000  # ubah ke jutaan
    
    elif "Juta" in first_price:
        value = float(first_price.replace("Juta", "").replace(",", ".").strip())
        return value
    
    return None

data["Harga Rumah"] = data["Harga Rumah"].apply(clean_harga)

In [ ]:
# CLEAN : LOKASI

data["Lokasi"] = data["Lokasi"].str.replace(", Bekasi", "")

In [68]:
# CARPORT : MISSING > 0 

data["Carport"] = data["Carport"].fillna(0)
data["Carport"] = data["Carport"].astype(int)

In [66]:
# CLEAN : JUMLAH KAMAR TIDUR

def clean_kamar(value):
    if pd.isna(value):
        return 0
    
    value = str(value)
    
    if "+" in value:
        parts = value.split("+")
        return int(parts[0]) + int(parts[1])
    else:
        return int(value)

data["Jumlah Kamar Tidur"] = data["Jumlah Kamar Tidur"].apply(clean_kamar)
data["Jumlah Kamar Mandi"] = data["Jumlah Kamar Mandi"].apply(clean_kamar)

In [73]:
# CLEAN : LUAS TANAH & LUAS BANGUNAN

data["Luas Tanah"] = data["Luas Tanah"].astype(float)
data["Luas Bangunan"] = data["Luas Bangunan"].astype(float)

In [69]:
# CHECK 
data.info()
data.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 7 columns):
 #   Column              Non-Null Count  Dtype  
---  ------              --------------  -----  
 0   Lokasi              200 non-null    object 
 1   Luas Tanah          200 non-null    float64
 2   Luas Bangunan       200 non-null    float64
 3   Carport             200 non-null    int64  
 4   Jumlah Kamar Tidur  200 non-null    int64  
 5   Jumlah Kamar Mandi  200 non-null    int64  
 6   Harga Rumah         200 non-null    float64
dtypes: float64(3), int64(3), object(1)
memory usage: 11.1+ KB


,Lokasi,Luas Tanah,Luas Bangunan,Carport,Jumlah Kamar Tidur,Jumlah Kamar Mandi,Harga Rumah
0,Premier,144.0,200.0,1,6,4,2200.0
1,Premier,105.0,120.0,1,0,0,1200.0
2,Premier,160.0,215.0,6,4,0,6800.0
3,Premier,112.0,156.0,4,4,0,4600.0
4,Premier,84.0,87.0,3,2,0,2300.0


### Insight after Data Cleaning

- Data terdiri dari 200 baris dan 7 kolom.
- Semua missing value sudah ditangani.
- Kolom numerik seperti Luas Tanah, Luas Bangunan, Carport, Jumlah Kamar Tidur, Jumlah Kamar Mandi, dan Harga Rumah sudah diubah ke tipe data numerik.
- Format harga sudah dikonversi ke dalam juta dan simbol mata uang sudah dihapus.
- Format luas tanah dan luas bangunan sudah dihilangkan satuannya sehingga bisa digunakan untuk perhitungan.
- Kolom lokasi sudah dibersihkan dengan menghapus kata ", Bekasi".

Dataset sekarang sudah bersih dan siap digunakan untuk selanjutnya : analisis statistik atau visualisasi.

In [74]:
# Save to CSV

data.to_csv("P0G3_data_ROSALIA.csv", index=False)

# C. Business Understanding/Problem Statement
Buatlah Problem Statement menggunakan SMART framework (berikut dengan penjabaran Specific, Measurable, Achievable, Relevant, dan Time-bound). Buatlah Problem Statement ke dalam satu kalimat dan gunakan metrik yang tepat pada kasus ini.

### 1. Specific

Analisis ini bertujuan untuk mengetahui faktor-faktor apa saja yang paling mempengaruhi harga rumah di Bekasi, seperti luas tanah, luas bangunan, jumlah kamar tidur, jumlah kamar mandi, dan carport, sehingga bisa diketahui variabel mana yang paling berdampak terhadap kenaikan harga rumah.

Metrics: mencari faktor paling berpengaruh terhadap harga.

### 2. Measurable

Pengaruh masing-masing variabel akan diukur menggunakan nilai korelasi dan statistik deskriptif (rata-rata, minimum, maksimum), sehingga dapat diketahui variabel mana yang memiliki hubungan paling kuat terhadap harga rumah.

Metrics: nilai korelasi & statistik numerik.

### 3. Achievable

Analisis ini dapat dilakukan karena tersedia 200 data listing rumah yang sudah melalui proses cleaning dan sudah dalam format numerik, sehingga siap digunakan untuk perhitungan statistik.

Metrics: bisa dikerjakan dengan data yang ada.

### 4. Relevant

Hasil analisis ini relevan untuk memahami pola harga properti di Bekasi dan dapat membantu pembeli dalam mempertimbangkan faktor utama yang mempengaruhi nilai rumah.

Metrics: Ada kaitan dengan dunia nyata (bukan cuma tugas).

### 5. Time-bound

Seluruh proses analisis, mulai dari eksplorasi data hingga penarikan kesimpulan, akan diselesaikan dalam waktu 1 minggu (teori dalam 3 hari, pembelajaran ulang & praktek coding dalam 4 hari)

Metrics: batas waktu dikerjakan sesuai deadline. 

## Summary
Analisis ini dilakukan untuk mengidentifikasi faktor-faktor seperti luas tanah, luas bangunan, jumlah kamar tidur, jumlah kamar mandi, dan carport yang berpengaruh terhadap harga rumah di Bekasi. Pengukuran dilakukan menggunakan statistik deskriptif dan analisis korelasi berdasarkan 200 data listing yang sudah dibersihkan dan diubah ke format numerik. Melalui analisis ini, diharapkan dapat diperoleh gambaran mengenai faktor utama yang memengaruhi harga properti di Bekasi, dengan waktu penyelesaian dalam satu periode pengerjaan tugas.

## D. Analysis
### 1. Dalam melakukan analisa data untuk mencapai tujuan yang diinginkan, Anda perlu mengikuti soal/pertanyaan/instruksi berikut ini :

Hitung rata-rata, median, standar deviasi, skewness, dan kurtosis dari semua kolom numerik. Dari hasil perhitungan, insight apa saja yang bisa Anda dapatkan (seperti distribusi data, kecenderungan ada/tidaknya outlier, dll)?

Note : Jika menemukan adanya indikasi outlier dari perhitungan ini, tidak perlu di-handle karena sifatnya alami. Dibiarkan saja.

In [76]:
# Rata-rata
mean_values = data.mean(numeric_only=True)

# Median
median_values = data.median(numeric_only=True)

# Standar Deviasi
std_values = data.std(numeric_only=True)

# Skewness
skew_values = data.skew(numeric_only=True)

# Kurtosis
kurtosis_values = data.kurtosis(numeric_only=True)

print(mean_values)
print(median_values)
print(std_values)
print(skew_values)
print(kurtosis_values)

Luas Tanah             134.755
Luas Bangunan          126.315
Carport                  1.860
Jumlah Kamar Tidur       1.105
Jumlah Kamar Mandi       0.285
Harga Rumah           1712.485
dtype: float64
Luas Tanah              84.0
Luas Bangunan           90.0
Carport                  2.0
Jumlah Kamar Tidur       0.0
Jumlah Kamar Mandi       0.0
Harga Rumah           1060.0
dtype: float64
Luas Tanah             239.582310
Luas Bangunan          172.509633
Carport                  1.093792
Jumlah Kamar Tidur       1.905816
Jumlah Kamar Mandi       1.019299
Harga Rumah           2223.095391
dtype: float64
Luas Tanah            8.789011
Luas Bangunan         9.189203
Carport               1.421776
Jumlah Kamar Tidur    1.612543
Jumlah Kamar Mandi    3.774940
Harga Rumah           6.162746
dtype: float64
Luas Tanah             85.575553
Luas Bangunan         106.871762
Carport                 3.551041
Jumlah Kamar Tidur      1.610728
Jumlah Kamar Mandi     14.758899
Harga Rumah            51

## Insights

### 1. Mean vs Median

Mean harga rumah (±1712 juta) lebih besar dari median (±1060 juta).
Artinya ada beberapa rumah dengan harga sangat tinggi yang membuat rata-rata naik.

Hal yang sama juga terlihat di luas tanah dan luas bangunan --> ada beberapa properti yang ukurannya jauh lebih besar dari kebanyakan.

### 2. Standar Deviasi

Standar deviasi harga rumah cukup besar.
Ini berarti harga rumah di dataset sangat bervariasi.
Luas tanah dan bangunan juga cukup bervariasi.

### 3. Skewness

Nilai skewness cenderung positif.
Artinya distribusi data miring ke kanan (right-skewed), yang berarti ada beberapa nilai besar (rumah mahal / besar).

### 4. Kurtosis

Nilai kurtosis cukup tinggi, terutama pada harga rumah.
Ini menunjukkan kemungkinan adanya nilai ekstrim (outlier), tapi ini wajar dalam data yang diinstruksikan dan tidak perlu dihapus.

## Kesimpulan 
Secara keseluruhan, data harga rumah dan ukuran properti memiliki variasi yang cukup besar. Rata-rata yang lebih tinggi dari median menunjukkan adanya beberapa rumah dengan harga atau ukuran yang jauh lebih besar dari mayoritas data. Distribusi data cenderung miring ke kanan (right-skewed) dan terdapat indikasi outliers, terutama pada harga rumah. Namun, kondisi ini masih wajar dalam dataset dan data sudah siap digunakan untuk analisis lanjutan.

### 2. Anda perlu mempersiapkan uang untuk pembelian rumah ini sehingga Anda harus mengetahui berapa kisaran harga rumah yang wajar di Bekasi ?

*Gunakan confidence interval untuk mendapatkan lower value dan upper value dari distribusi populasi harga rumah. Gunakan confidence level 95%.

In [77]:
harga = data["Harga Rumah"]

mean = harga.mean()
std = harga.std()
n = len(harga)

# Standard Error
se = std / (n ** 0.5)

# Confidence Interval 95%
lower = mean - 1.96 * se
upper = mean + 1.96 * se

print("Mean Harga Rumah:", mean)
print("Lower Bound (95%):", lower)
print("Upper Bound (95%):", upper)

Mean Harga Rumah: 1712.485
Lower Bound (95%): 1404.3796981311802
Upper Bound (95%): 2020.5903018688196


### Insights

Dengan menggunakan confidence level 95%, rata-rata harga rumah di Bekasi diperkirakan berada di antara 1404 juta (lower bound) hingga 2020 juta (upper bound). Selain itu, nilai rata-rata harga rumah adalah sekitar Rp 1.7 Miliar. 

### Kesimpulan : 
Ini berarti, kisaran harga rumah yang wajar di Bekasi berada di sekitar Rp 1,4 Miliar - Rp 2 Miliar
- Jadi, pelanggan bisa menargetkan harga rumah di Rp 1.7 Miliar untuk optimalisasi harga yang terbaik.


### 3. Saat ini Anda belum memiliki mobil sehingga Carport bukan prioritas saat ini. Anda mau tahu apakah rumah dengan Carport lebih mahal dibandingkan tanpa Carport. Berikan analisa apakah layak membayar lebih untuk rumah yang memiliki Carport? 

Gunakan uji hipotesis yang diawali dengan menuliskan hipotesis null dan alternatifnya serta tentukan jenis hipotesis yang digunakan.

In [ ]:
from scipy.stats import ttest_ind

# Pisahkan harga berdasarkan carport
harga_carport = data[data["Carport"] > 0]["Harga Rumah"]
harga_tanpa_carport = data[data["Carport"] == 0]["Harga Rumah"]

# Independent t-test
t_stat, p_value_two_tailed = ttest_ind(harga_carport, harga_tanpa_carport)

# Because this is one-tailed test:
p_value_one_tailed = p_value_two_tailed / 2

print("T-statistic:", t_stat)
print("P-value (two-tailed):", p_value_two_tailed)
print("P-value (one-tailed):", p_value_one_tailed)

T-statistic: -7.165862630330822
P-value (two-tailed): 1.49274280190768e-11
P-value (one-tailed): 7.4637140095384e-12


## Insight : 

Hasil : 
T-statistic = -7.17
P-value (one-tailed) ≈ 7.46e-12 

P-value sangat kecil, jauh < 0.05, Tapi t-statistic NEGATIF

Maka : 
Rata-rata harga rumah dengan carport lebih rendah daripada tanpa carport. 

## Kesimpulan : 

Berdasarkan hasil analisa diatas, tidak ada bukti bahwa rumah dengan carport lebih mahal. Bahkan, rata-ratanya cenderung lebih rendah. Jadi secara statistik, tidak ada indikasi bahwa carport membuat harga rumah lebih tinggi. Dengan demikian, pelanggan tidak layak untuk membayar lebih mahal dengan adanya carport. 

Jika harus memilih, apakah Anda lebih baik membeli tanah yang luas ataukah membeli bangunan yang luas ? Manakah yang lebih mempengaruhi harga rumah, apakah Luas Tanah ataukah Luas Bangunan ? Berikan pendapatmu.

Anda dapat jawab pertanyaan ini dengan uji korelasi. Gunakan library SciPy (bukan Pandas). Analisis nilai korelasi dan p-value nya. Gunakan teknik yang sesuai dengan kondisi data !

### 4 . Jika harus memilih, apakah Anda lebih baik membeli tanah yang luas ataukah membeli bangunan yang luas ? Manakah yang lebih mempengaruhi harga rumah, apakah Luas Tanah ataukah Luas Bangunan ? Berikan pendapatmu.

Anda dapat jawab pertanyaan ini dengan uji korelasi. Gunakan library SciPy (bukan Pandas). Analisis nilai korelasi dan p-value nya. Gunakan teknik yang sesuai dengan kondisi data !

In [79]:
from scipy.stats import pearsonr

# Luas Tanah & Harga
corr_tanah, p_tanah = pearsonr(data["Luas Tanah"], data["Harga Rumah"])

# Luas Bangunan & Harga
corr_bangunan, p_bangunan = pearsonr(data["Luas Bangunan"], data["Harga Rumah"])

print("Korelasi Luas Tanah vs Harga:", corr_tanah)
print("P-value Luas Tanah:", p_tanah)

print("Korelasi Luas Bangunan vs Harga:", corr_bangunan)
print("P-value Luas Bangunan:", p_bangunan)

Korelasi Luas Tanah vs Harga: 0.8937534475434061
P-value Luas Tanah: 7.269192239221243e-71
Korelasi Luas Bangunan vs Harga: 0.8435125677000451
P-value Luas Bangunan: 2.390686588892007e-55


### Insight 

Kedua korelasi cukup tinggi, artinya : 
- semakin luas tanah, harga rumah semakin tinggi.
- semakin luas bangunan, harga rumah juga meningkat.

Kita bisa memakai dari nilai korelasi : 
- Luas Tanah = 0.894
- Luas Bangunan = 0.844

--> luas tanah sedikit lebih tinggi dalam menentukan harga.

### Kesimpulan 

Berdasarkan hasil uji korelasi Pearson, baik Luas Tanah maupun Luas Bangunan sama-sama punya hubungan positif yang sangat kuat dengan harga rumah, dan hasilnya juga signifikan secara statistik (p-value < 0.05). Tapi kalau dibandingkan, korelasi Luas Tanah (0.894) sedikit lebih tinggi daripada Luas Bangunan (0.844).

Jadi nilai tanah sedikit mempengaruhi. 
Sebagai pilihan, membeli tanah yang lebih luas akan memberikan dampak yang lebih besar terhadap nilai properti.

## E. Conclusion
Please check each markdown above, each conclusion has been given. Thank you!